### ISSUE TITLE
ER: manageability/status functions for lms2012

### ISSUE THREAD

**Issue by edleft** [ev3dev/lms2012-compat#73]
Enhancement request: it'd be nice to be able to query lms2012 about its status and perform some management functions. This would lend itself to a use case of keeping it running as a daemon, so that you can connect to it with EVG but also switch to runtime mode immediately to run and manage programs. As it stands now, I tend to either start it standalone to do some EVG functions or launch lms2012 with a path to an .rbf to run a program. And when doing the latter, there's no visibility into the running program--it's fire and forget or requires hacking into the Lego programs some crude communication via files of status/etc.

The functions most useful:
0. basic health check (everything ok, able to respond at app level, not hung)
1. stop currently running program
2. get currently running program (<project>/<program>.rbf) and currently running block

Nice to have:
3. get system start time of lms2012 and of program.rbf that is running
4. ask lms2012 to shutdown gracefully or to reinit/restart itself

**Comment by dlech** [comment]
So basically, you are thinking some kind of command line that will communicate with a running instance?

> 3. get system start time of lms2012 ~and of program.rbf that is running~
> 4. ask lms2012 to shutdown gracefully or to reinit/restart itself

These (minus the crossed out part) are easily accomplished with a systemd service.

    sudo systemctl {start,restart,stop,status} lms2012

The rest could be handled with a separate program that sends direct commands to the running lsm2012 instance. For example, [PROGRAM_INFO](https://github.com/ev3dev/lmsasm/blob/master/bytecodes/data/ev3.yml#L1119) looks interesting.

**Comment by edleft** [comment]
command line would be great, as it'd have some structure for feedback as well as automation, plus be easily shell-accessible and be fairly concise.

yes, some wrappers around the current EV3 system commands would be great.

**Comment by edleft** [comment]
yes, some wrappers around the current system commands would be great.
